Source/Core/EllipsoidRhumbLine.js

```diff
@@ -0,0 +1,431 @@
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('start', start);
+        Check.defined('heading', heading);
+        Check.defined('distance', distance);
+        //>>includeEnd('debug');
+
+        var e = defaultValue(ellipsoid, Ellipsoid.WGS84);
+
+        if (defined(result)) {
+            result._ellipsoid = e;
+            initiailize(result, start, undefined, e);
+            result._heading = CesiumMath.negativePiToPi(heading);
+            result._distance = distance;
+
+            result._end = result.interpolateUsingSurfaceDistance(distance, new Cartographic());
+            return result;
+        }
+
+        var rhumbLine = new EllipsoidRhumbLine(undefined, undefined, e);
+        initiailize(rhumbLine, start, undefined, e);
+        rhumbLine._heading = CesiumMath.negativePiToPi(heading);
+        rhumbLine._distance = distance;
+        rhumbLine._end = rhumbLine.interpolateUsingSurfaceDistance(distance, new Cartographic());
+
+        return rhumbLine;
+    };
+
+    /**
+     * Sets the start and end points of the rhumb line.
+     *
+     * @param {Cartographic} start The initial planetodetic point on the path.
+     * @param {Cartographic} end The final planetodetic point on the path.
+     */
+    EllipsoidRhumbLine.prototype.setEndPoints = function(start, end) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('start', start);
+        Check.defined('end', end);
+        //>>includeEnd('debug');
+
+        computeProperties(this, start, end, this._ellipsoid);
+    };
+
+    /**
+     * Provides the location of a point at the indicated portion along the rhumb line.
+     *
+     * @param {Number} fraction The portion of the distance between the initial and final points.
+     * @param {Cartographic} result The object in which to store the result.
+     * @returns {Cartographic} The location of the point along the rhumb line.
+     */
+    EllipsoidRhumbLine.prototype.interpolateUsingFraction = function(fraction, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.typeOf.number('distance', this._distance);
+        //>>includeEnd('debug');
+
+        return this.interpolateUsingSurfaceDistance(fraction * this._distance, result);
+    };
+
+    /**
+     * Provides the location of a point at the indicated distance along the rhumb line.
+     *
+     * @param {Number} distance The distance from the inital point to the point of interest along the rhumbLine.
+     * @param {Cartographic} result The object in which to store the result.
+     * @returns {Cartographic} The location of the point along the rhumb line.
+     *
+     * @exception {DeveloperError} start and end must be set before calling function interpolateUsingSurfaceDistance
+     */
+    EllipsoidRhumbLine.prototype.interpolateUsingSurfaceDistance = function(distance, result) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('distance', this._distance);
```

Comment:
Based on the first couple specs for `EllipsoidRhumb` I'm not sure if this check will produce the most helpful error message - users might not realize that they need to set `start` and `end` to get a `distance`. I guess this is how `EllipsoidGeodesic` works, but any thoughts on an alternative?

Reply:
Yeah I wanted to keep it consistent with `EllipsoidGeodesic` and hence similar checks. I think it's fine to keep them as is for now.